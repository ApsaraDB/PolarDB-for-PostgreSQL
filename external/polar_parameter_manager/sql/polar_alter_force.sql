-- TEST ALTER FORCE
set client_min_messages to error; -- set client_min_messages to avoid other session reload_conf and cause unstable case
-- ## invalid ALTER FORCE ##
-- non-polar guc
alter system for cluster reload force set xxx to 'xxx';
alter system force set xxx to 'xxx';
alter system reload force set xxx to 'xxx';
alter system for cluster force set xxx to 'xxx';
-- value contain a newline
alter system for cluster force set polar_xxx to 'xxx
';
alter system for cluster reload force set xxx to '
xxx
';
alter system force set xxx to 'x
xx';
alter system reload force set xxx to '
xxx';
alter system for cluster force set xxx to 'x
xx';
-- ## valid ALTER FORCE ##
-- accept polar guc
alter system force set polar_xxx to 'xxx';
alter system force reset polar_xxx;
alter system for cluster force set polar_xxx to 'xxx';
alter system for cluster force reset polar_xxx;
-- accept extension guc
alter system force set xx1.xxx to 'xxx';
alter system force set polar_xx1.xxx to 'xxx';
alter system for cluster force set xx2.xxx to 'xxx';
alter system for cluster force set polar_xx2.xxx to 'xxx';
-- check force alter success
select count(distinct name) from pg_file_settings where
name in ('xx1.xxx','xx2.xxx','polar_xx1.xxx','polar_xx2.xxx');
-- reset extension guc at last.
alter system force reset xx1.xxx;
alter system force reset polar_xx1.xxx;
alter system for cluster force reset xx2.xxx;
alter system for cluster force reset polar_xx2.xxx;
-- check force alter reset success
select count(distinct name) from pg_file_settings where
name in ('xx1.xxx','xx2.xxx','polar_xx1.xxx','polar_xx2.xxx');
-- exists guc
alter system force set debug_print_parse to on;
alter system force reset debug_print_parse;
alter system for cluster force set debug_print_parse to on;
alter system for cluster force reset debug_print_parse;
-- exists guc sill check value
alter system force set debug_print_parse to invalid;
alter system for cluster force set debug_print_parse to invalid;
reset client_min_messages;
