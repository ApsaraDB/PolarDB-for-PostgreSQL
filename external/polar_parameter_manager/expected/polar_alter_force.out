-- TEST ALTER FORCE
set client_min_messages to error; -- set client_min_messages to avoid other session reload_conf and cause unstable case
-- ## invalid ALTER FORCE ##
-- non-polar guc
alter system for cluster reload force set xxx to 'xxx';
ERROR:  invalid configuration parameter "xxx"
HINT:  force set only support polar and extension parameter
alter system force set xxx to 'xxx';
ERROR:  invalid configuration parameter "xxx"
HINT:  force set only support polar and extension parameter
alter system reload force set xxx to 'xxx';
ERROR:  invalid configuration parameter "xxx"
HINT:  force set only support polar and extension parameter
alter system for cluster force set xxx to 'xxx';
ERROR:  invalid configuration parameter "xxx"
HINT:  force set only support polar and extension parameter
-- value contain a newline
alter system for cluster force set polar_xxx to 'xxx
';
ERROR:  parameter value for ALTER SYSTEM must not contain a newline
alter system for cluster reload force set xxx to '
xxx
';
ERROR:  parameter value for ALTER SYSTEM must not contain a newline
alter system force set xxx to 'x
xx';
ERROR:  parameter value for ALTER SYSTEM must not contain a newline
alter system reload force set xxx to '
xxx';
ERROR:  parameter value for ALTER SYSTEM must not contain a newline
alter system for cluster force set xxx to 'x
xx';
ERROR:  parameter value for ALTER SYSTEM must not contain a newline
-- ## valid ALTER FORCE ##
-- accept polar guc
alter system force set polar_xxx to 'xxx';
alter system force reset polar_xxx;
alter system for cluster force set polar_xxx to 'xxx';
alter system for cluster force reset polar_xxx;
-- accept extension guc
alter system force set xx1.xxx to 'xxx';
alter system force set polar_xx1.xxx to 'xxx';
alter system for cluster force set xx2.xxx to 'xxx';
alter system for cluster force set polar_xx2.xxx to 'xxx';
-- check force alter success
select count(distinct name) from pg_file_settings where
name in ('xx1.xxx','xx2.xxx','polar_xx1.xxx','polar_xx2.xxx');
 count 
-------
     4
(1 row)

-- reset extension guc at last.
alter system force reset xx1.xxx;
alter system force reset polar_xx1.xxx;
alter system for cluster force reset xx2.xxx;
alter system for cluster force reset polar_xx2.xxx;
-- check force alter reset success
select count(distinct name) from pg_file_settings where
name in ('xx1.xxx','xx2.xxx','polar_xx1.xxx','polar_xx2.xxx');
 count 
-------
     0
(1 row)

-- exists guc
alter system force set debug_print_parse to on;
alter system force reset debug_print_parse;
alter system for cluster force set debug_print_parse to on;
alter system for cluster force reset debug_print_parse;
-- exists guc sill check value
alter system force set debug_print_parse to invalid;
ERROR:  parameter "debug_print_parse" requires a Boolean value
alter system for cluster force set debug_print_parse to invalid;
ERROR:  parameter "debug_print_parse" requires a Boolean value
reset client_min_messages;
